test/Window/WindowBase.test.cpp

```diff
@@ -219,5 +221,30 @@ TEST_CASE("[Window] sf::WindowBase", runDisplayTests())
 
         // Should compile if user provides both a specific handler and a catch-all
         windowBase.handleEvents([](const sf::Event::Closed&) {}, [](const auto&) {});
+
+        // Should compile if user provides a handler taking an event subtype by value or reference,
+        // but not rvalue reference because it would never be called.
+        windowBase.handleEvents([](sf::Event::Closed) {});
+        windowBase.handleEvents([](const sf::Event::Closed) {});
+        windowBase.handleEvents([](sf::Event::Closed&) {});
+        windowBase.handleEvents([](const sf::Event::Closed&) {});
+
+        // Should compile if user provides a handler with deleted rvalue ref-qualified call operator
+        struct LvalueOnlyHandler
+        {
+            void operator()(const sf::Event::Closed&) &
+            {
+            }
+            void operator()(const sf::Event::Closed&) && = delete;
+        };
+        windowBase.handleEvents(LvalueOnlyHandler{});
```

Comment:
This shouldn't compile though you are passing an rvalue to it.

You should test when the handlerItself is an Lvalue


```suggestion
LvalueOnlyHandler handler; windowBase.handleEvents(handler);
```

Reply:
The thing is this rvalue is used to initialize the OverloadSet that is then used as an lvalue to visit events.

Reply:
> The thing is this rvalue is used to initialize the OverloadSet that is then used as an lvalue to visit events.

so OverloadSet doesn't preserve reference category is that wanted?

Reply:
I think so, because it is used potentially many times in the event loop.